### DISCUSSION TITLE
Xiaomi  LYWSD02MMC BLE Support

### DISCUSSION THREAD

**Discussion by DaniilMir** [arendst/Tasmota#19415] [Q&A]
Hello Community. 
I was using a smaller version lywsd03mmc and decided to switch to a bgger version LYWSD02MMC thinking that it should be an easy task.
The lywsd03mmc  was working fine on the official bluetooth build. 
When I added the LYWSD02MMC with the extracted key - it wasn't working at all. 
I'm getting the following errors: 
```
No Key found
Payload encrypted but no key
```
I was able to make it work using the ESP Home and this [component](https://github.com/dentra/esphome-components/tree/master/components/miot_th)

Befor opening this discussion I found an apandoned topic wih no helpful info. 
Is there a way to build a tasmota firmware with support for LYWSD02MMC sensor? 

**P.S.** Key is working fine when using with ESP Home or Homeassistant integration. Issue is with tasmota firmware only.

**Comment by Jason2866** [comment]
Try this one https://github.com/tasmota/install/blob/firmware/firmware/unofficial/tasmota32-mi32-homebridge.bin
The device is supported via the MI driver NOT the Bluetooth build. https://tasmota.github.io/docs/Bluetooth_MI32/

**Reply by Staars** [reply]
No, I fear this a newer sensor with added encryption, that has not been adeed to any of Tasmotas drivers.
Back in the days when Bluetooth development started on the ESP8266 with an external BLE device there was only 1 known Xiaomi sensor and then the list of potential devices grew through the roof.
With today’s knowledge and the ESP32 as primary working platform it would probably make sense to completely rework the Xiaomi BLE section of Tasmota with Berry and usage of config files in order to add new sensors more easily.

**Reply by DaniilMir** [reply]
Hello @Jason2866 
Thanks for sharing an alternative solution. 
I've tried it on weekend, it was able to read my old re-flashed sensors. 
New Xiaomi sensors are failing to decrypt with the following error: 

> 13:14:23.809 M32: new MI_2542 at slot: 2
> 13:14:23.809 M32: Decryption failed with error: -2

Key was added directly in console with the "MI32key" command. 
Maybe this thermomether is not supported in Tasmota
